src/util/file_reader.js

```diff
@@ -115,8 +115,8 @@ class NewlineReader {
     async init() {
         let fh = null;
         try {
-            fh = await nb_native().fs.open(this.fs_context, this.path, 'r');
-            if (this.lock) await fh.flock(this.fs_context, this.lock);
+            fh = await nb_native().fs.open(this.fs_context, this.path, '+');
```

Comment:
is RW needed for fcntl locking?

Reply:
From man page:
```
       EBADF  fd is not an open file descriptor

       EBADF  op is F_SETLK or F_SETLKW and the file descriptor open
              mode doesn't match with the type of lock requested.
```

The second one causes a EBADF if the descriptor is not "compatible". That's why I had to change permissions at a few places.

Reply:
so is O_RDRW only required only for exclusive lock, and O_RDONLY can work with shared? 
just trying to understand the limitations, because in that scope we are just trying the read the file...

Reply:
Ok - so I would put a code comment to explain that we open RW for this reading flow because the need for exclusive locking